lib/fiddle/ffi_backend.rb

```diff
@@ -323,7 +323,7 @@ def initialize(addr, size = nil, free = nil)
               FFI::Pointer.new(Integer(addr))
             end
 
-      @size = size ? size : ptr.size
+      @size = size ? size : (ptr.size_limit? ? ptr.size : 0)
```

Comment:
Nested `? :` may be difficult to read. How about using `if/elsif` instead?

```suggestion
      if size
        @size = size
      elsif ptr.size_limit?
        @size = ptr.size
      else
        @size = 0
      end
```